### ISSUE TITLE
Type resolution cannot resolve generated classes (BuildConfig, R) under AGP built-in Kotlin

### ISSUE THREAD

**Issue by michnovka** [detekt/detekt#9402]
## Expected Behavior

Under AGP built-in Kotlin, the type-resolution tasks (`detektMain` / `detektDebug`) should resolve references to generated classes (`BuildConfig`, `R`, view binding, KSP/protobuf output) — the same as with the legacy `kotlin-android` plugin.

## Observed Behavior

detekt logs `There were N compiler errors found during analysis` with `unresolved reference 'BuildConfig'` (etc.), and type-resolution rules then misfire (e.g. false `UnusedPrivateFunction` on used functions, `RedundantSuspendModifier` on real suspend calls).

Since #9251, generated sources are excluded from the analyzed source set — but their compiled classes are not added to the classpath either, so generated types become unresolvable.

## Steps to Reproduce

Minimal Android library, AGP 9 built-in Kotlin (no `kotlin-android`), detekt `2.0.0-alpha.4`:

```kotlin
// build.gradle.kts
plugins {
    id("com.android.library") version "9.2.1"
    id("dev.detekt") version "2.0.0-alpha.4"
}
android {
    namespace = "com.example"
    compileSdk = 36
    buildFeatures { buildConfig = true }
}
```

```kotlin
// src/main/kotlin/Sample.kt
class Sample {
    fun isDebug() = BuildConfig.DEBUG
}
```

```
$ ./gradlew detektMain
e: .../Sample.kt: error: unresolved reference 'BuildConfig'.
There were 1 compiler errors found during analysis. This affects accuracy of reporting.
```

The same code analyzes cleanly with `id("org.jetbrains.kotlin.android")` + `android.builtInKotlin=false`, `android.newDsl=false`.

## Context

Real modules (protobuf/KSP/many generated types) hit hundreds of these errors, producing false positives that fail the build. The only workaround is keeping the legacy `kotlin-android` plugin + those two flags, all of which are removed in AGP 10. I have a fix and will open a PR.

## Your Environment

* Version of detekt used: `2.0.0-alpha.4`
* Version of Gradle used: `9.5.1`
* AGP version: `9.2.1`
* Operating System and version: Linux

**Comment by cortinico** [comment]
Is this a regression from alpha.3? I suppose yes but just checking.

**Comment by michnovka** [comment]
Yeah, you're right, it goes back to alpha.3, the first release with built-in Kotlin support (#9100). I checked a small built-in Kotlin app with 2.0.0-alpha.3 and BuildConfig/ViewBinding are already unresolved there (6 compiler errors found during analysis), so it's been there since the feature landed rather than a later regression. The generated classes just never made it onto the analysis classpath under built-in Kotlin.

**Comment by FletchMcKee** [comment]
I wonder if this change needs to be made:
```diff
- val kotlin = sources.kotlin?.all ?: return
+ val kotlin = sources.java?.all ?: return
val source = project.objects.fileCollection().from(kotlin)
```

I’ll look into this because the regression is likely from #9251 .

A temporary fix you can try is adding the build generated folder as a source to the Detekt task:
```kotlin
 tasks.withType<dev.detekt.gradle.Detekt>().configureEach {
   source(project.layout.buildDirectory.dir("generated"))
 }
```

**Comment by michnovka** [comment]
Thanks! I ran all of these down empirically (patched the plugin for each variant and checked detekt's actual `--input`/`--classpath` and output), and the source-based ones don't resolve it.

The root cause is that detekt's CLI only feeds `.kt` files to `--input`, so generated Java like `BuildConfig` is never analyzed from source. It can only be resolved as a compiled class on `--classpath`, which is what this PR adds, and which keeps generated code out of analysis as a bonus, so it stays in line with #9251 rather than undoing it.

What I saw:
- `sources.java?.all` drops `src/main/kotlin` entirely (no `.kt` files in it), so detekt analyzes no Kotlin at all. Move a `.kt` that references `BuildConfig` into `src/main/java` and it still fails with "compiler errors found during analysis", even with the generated `BuildConfig.java` on `--input`.
- `source(build/generated)` doesn't even get that far, Gradle rejects it for consuming the other variant's output.

On the regression, I don't think it's #9251. I built detekt at its parent commit (still `setSource(siblingTask.sources)`) and `BuildConfig` is unresolved there too, so it predates #9251 and traces back to #9100, where the generated classes were just never on the analysis classpath.

**Comment by FletchMcKee** [comment]
Ah thanks for that additional information. Android's builtInKotlin change has introduced a lot of Gradle headaches.
